src/modules/auth/auth.controller.spec.ts

```diff
@@ -0,0 +1,253 @@
+import { newUser } from './../../../test/fixtures';
+import { AuthController } from './auth.controller';
+import { UserUseCases } from '../user/user.usecase';
+import { KeyServerUseCases } from '../keyserver/key-server.usecase';
+import { CryptoService } from '../../externals/crypto/crypto.service';
+import { LoginDto } from './dto/login-dto';
+import { LoginAccessDto } from './dto/login-access.dto';
+import { Response } from 'express';
+import {
+  BadRequestException,
+  ConflictException,
+  NotFoundException,
+} from '@nestjs/common';
+import { KeyServer } from '../keyserver/key-server.domain';
+import { DeepMocked, createMock } from '@golevelup/ts-jest';
+import { v4 } from 'uuid';
+import { TwoFactorAuthService } from './two-factor-auth.service';
+import { GeneratedSecret } from 'speakeasy';
+import { UpdateTfaDto } from './dto/update-tfa.dto';
+import { DeleteTfaDto } from './dto/delete-tfa.dto';
+
+describe('AuthController', () => {
+  let authController: AuthController;
+  let userUseCases: DeepMocked<UserUseCases>;
+  let keyServerUseCases: DeepMocked<KeyServerUseCases>;
+  let cryptoService: DeepMocked<CryptoService>;
+  let twoFactorAuthService: DeepMocked<TwoFactorAuthService>;
+
+  beforeEach(async () => {
+    userUseCases = createMock<UserUseCases>();
+    keyServerUseCases = createMock<KeyServerUseCases>();
+    cryptoService = createMock<CryptoService>();
+    twoFactorAuthService = createMock<TwoFactorAuthService>();
+
+    authController = new AuthController(
+      userUseCases,
+      keyServerUseCases,
+      cryptoService,
+      twoFactorAuthService,
+    );
+  });
+
+  it('should be defined', () => {
+    expect(authController).toBeDefined();
+  });
+
+  describe('POST /login', () => {
+    it('When valid credentials are provided, then it should return security details', async () => {
+      const clientId = 'drive-mobile';
+      const loginDto = new LoginDto();
+      loginDto.email = 'test@example.com';
+
+      const user = newUser();
+      user.hKey = 'hKey';
+      user.secret_2FA = 'secret_2FA';
+
+      const keys: Omit<
+        KeyServer,
+        'id' | 'userId' | 'encryptVersion' | 'toJSON'
+      > = {
+        publicKey: 'publicKey',
+        privateKey: 'privateKey',
+        revocationKey: 'revocationKey',
+      };
+
+      jest.spyOn(userUseCases, 'findByEmail').mockResolvedValueOnce(user);
+      jest.spyOn(keyServerUseCases, 'findUserKeys').mockResolvedValueOnce(keys);
+      jest.spyOn(cryptoService, 'encryptText').mockReturnValue('encryptedText');
+
+      const res = {
+        status: jest.fn(() => res),
+        send: jest.fn(),
+      } as unknown as Response;
+
+      const result = await authController.login(loginDto, clientId);
+
+      expect(result).toEqual({
+        hasKeys: keys,
+        sKey: 'encryptedText',
+        tfa: true,
+      });
+    });
+
+    it('When user is not found, then it should throw NotFoundException', async () => {
```

Comment:
This test is wrong, the current login does not [behave](https://github.com/internxt/drive-server/blob/f6dbe3faab26a68637661083e893607854f66fd8/src/app/routes/auth.ts#L78) like that. On the other hand, we should not inform about if a user exists or not in our service, that allows enumeration attacks to be run.